MainModule/Server/Core/Process.lua

```diff
@@ -384,14 +391,20 @@ return function(Vargs)
 						Player = p;
 					})
 
-					service.Events.PlayerChatted:Fire(p,msg)
-
 					if Settings.ChatCommands then
-						if msg:sub(1,3)=="/e " then
+						if Remote.Get(p,"Function","ChatCmdsHidden") == true and (msg:sub(1,1) == Settings.Prefix or msg:sub(1,1) == Settings.PlayerPrefix) then
+							Remote.Send(p,"Function","ChatMessage","> "..msg,Color3.new(255, 255, 255))
+							Process.Command(p,msg,{Chat = true;})
+						elseif msg:sub(1,3)=="/e " then
+							service.Events.PlayerChatted:Fire(p,msg)
 							msg = msg:sub(4)
+							Process.Command(p,msg,{Chat = true;})
```

Comment:
I know this isn't part of your PR, but it might be worth allowing support for "/system "
```suggestion
							Process.Command(p,msg,{Chat = true;})
						elseif msg:sub(1,8)=="/system " then
							service.Events.PlayerChatted:Fire(p,msg)
							msg = msg:sub(9)
							Process.Command(p,msg,{Chat = true;})
```